staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -490,3 +509,78 @@ func startDBSizeMonitorPerEndpoint(client *clientv3.Client, interval time.Durati
 		cancel()
 	}, nil
 }
+
+// checkVersion checks for deprecated versions of etcd and returns an error if find any
+// currently the versions < 3.4.25, between 3.5.0 and 3.5.7
+func checkVersion(c *clientv3.Client, dialTimeOut time.Duration) (err error) {
+	var wg sync.WaitGroup
+
+	eps := c.Endpoints()
+	errc := make(chan error, len(eps))
+	baseCtx := context.TODO()
+
+	ctx, cancel := context.WithCancel(baseCtx)
+	if dialTimeOut > 0 {
+		cancel()
+		ctx, cancel = context.WithTimeout(ctx, dialTimeout)
+	}
+
+	wg.Add(len(eps))
```

Comment:
Please consider using https://pkg.go.dev/golang.org/x/sync/errgroup to simplify error handling.